scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+	//item->setLineTransparency(1.0);
+}
+
+TextFramework::TextFramework()
+{
+	m_textRegions.push_back(activeTextRegion);
+	setCharMode(AddCharMode::ADDFIRSTCHAR);
+}
+
+TextFramework::~TextFramework()
+{
+}
+
+void TextFramework::addNewTextRegion()
+{
+	activeTextRegion = TextRegion();
+	m_textRegions.push_back(activeTextRegion);
+	setCharMode(TextFramework::AddCharMode::ADDFIRSTCHAR);
+}
+
+void TextFramework::addChar(GfxState* state, double x, double y, double dx, double dy, double originX, double originY, CharCode code, int nBytes, Unicode const* u, int uLen)
+{
+	(this->*(m_addChar))(state, x, y, dx, dy, originX, originY, code, nBytes, u, uLen);
+}
+
+bool TextFramework::isNewLineOrRegion(QPointF newPosition)
+{
+	return (activeTextRegion.coLinera(activeTextRegion.lastXY.y(), activeTextRegion.textRegionLines.back().baseOrigin.y()) &&
+		!activeTextRegion.coLinera(newPosition.y(), activeTextRegion.lastXY.y()))
+		|| (activeTextRegion.coLinera(newPosition.y(), activeTextRegion.lastXY.y())
+			&& !activeTextRegion.closeToX(newPosition.x(), activeTextRegion.lastXY.x()));
+}
+
+PdfGlyph TextFramework::AddFirstChar(GfxState* state, double x, double y, double dx, double dy, double originX, double originY, CharCode code, int nBytes, Unicode const* u, int uLen)
+{
+	//qDebug() << "AddFirstChar() '" << u << " : " << uLen;	
+	PdfGlyph newGlyph = TextFramework::AddCharCommon(state, x, y, dx, dy, u, uLen);
+	activeTextRegion.glyphs.push_back(newGlyph);
+	setCharMode(AddCharMode::ADDBASICCHAR);
+
+	//only need to be called for the very first point
+	if (activeTextRegion.addGlyphAtPoint(QPointF(x, y), newGlyph) == TextRegion::FrameworkLineTests::FAIL)
+	{
+		qDebug("FIXME: Rogue glyph detected, this should never happen because the coursor should move before glyphs in new regions are added.");
+	}
+	return newGlyph;
+}
+PdfGlyph TextFramework::AddCharCommon(GfxState* state, double x, double y, double dx, double dy, Unicode const* u, int uLen)
+{
+	//qDebug() << "AddBasicChar() '" << u << " : " << uLen;
+	PdfGlyph newGlyph;
+	newGlyph.dx = dx;
+	newGlyph.dy = dy;
+
+	// Convert the character to UTF-16 since that's our SVG document's encoding
+	for (int i = 0; i < uLen; i++)
+	{
+		newGlyph.code = (char16_t)u[i];
+	}
+
+	newGlyph.rise = state->getRise();
+	return newGlyph;
+}
+
+PdfGlyph TextFramework::AddBasicChar(GfxState* state, double x, double y, double dx, double dy, double originX, double originY, CharCode code, int nBytes, Unicode const* u, int uLen)
+{
+	PdfGlyph newGlyph = AddCharCommon(state, x, y, dx, dy, u, uLen);
+	activeTextRegion.lastXY = QPointF(x, y);
+	activeTextRegion.glyphs.push_back(newGlyph);
+	return newGlyph;
+}
+
+PdfGlyph TextFramework::AddCharWithNewStyle(GfxState* state, double x, double y, double dx, double dy, double originX, double originY, CharCode code, int nBytes, Unicode const* u, int uLen)

```

Comment:
i guess that this and the next function are not implemented yet... you should state this in the doc line

Reply:
ok